tests/integration/cluster_fixture.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import copy
+import os
+import threading
+from uuid import uuid4, uuid1
+
+from trivup.clusters.KafkaCluster import KafkaCluster
+
+from confluent_kafka import Producer, Consumer
+
+from confluent_kafka.admin import AdminClient, NewTopic
+
+
+class ClusterFixture(object):
+    _slots_ = ()
+    """
+    ClusterFixture factory.
+    Provides client configs enabling integration tests to execute against a live cluster.
+    """
+
+    def __new__(cls, options):
+        conf = {k[6:]: v for (k, v) in os.environ.items() if k.startswith('KAFKA')}
+
+        if bool(conf):
+            return ExternalClusterFixture(conf, options)
```

Comment:
Shouldn't External and Trivup ClusterFixture classes derive from the same base class, making it more obvious what methods they supply.
As it currently stands TrivupClusterFixture provides more methods than the ExternalClusterFixture (e.g., create_topic), which makes it unclear if that is a public method or not.

Not sure having a factory return incompatible types is practical.